python_modules/dagster/dagster/_core/definitions/definitions_class.py

```diff
@@ -0,0 +1,92 @@
+from typing import Any, Dict, Iterable, Mapping, Optional, Union
+
+from dagster._annotations import experimental
+from dagster._core.execution.with_resources import with_resources
+
+from .assets import AssetsDefinition, SourceAsset
+from .cacheable_assets import CacheableAssetsDefinition
+from .decorators import repository
+from .job_definition import JobDefinition
+from .repository_definition import (
+    SINGLETON_REPOSITORY_NAME,
+    PendingRepositoryDefinition,
+    RepositoryDefinition,
+)
+from .resource_definition import ResourceDefinition
+from .schedule_definition import ScheduleDefinition
+from .sensor_definition import SensorDefinition
+
+
+@experimental
+class Definitions:
+    def __init__(
+        self,
+        assets: Optional[
```

Comment:
Historically, we've allowed for nested lists to be returned from the repository function, to make it more ergonomic to do things like

```
assets = [
    load_assets_from_dbt_project(...), # returns a list
    some_asset,
    another_asset,
]
```

without having to splat everything out.

if you ignore the type here, I think that will still work with this class, but it would probably be good to either take the hard-line stance that this list should be pre-flattened or update the type.

Reply:
IMO we should just insist on pre-flattening to make the type signature reasonable